File: fantasy-draft-backend/src/main/java/com/catch23/fantasy_draft_backend/model/league/PlayerSettings.java
package com.catch23.fantasy_draft_backend.model;

public class PlayerSettings {
    // Enums
    public enum Division {
        AL, NL, MIXED
    }

    public enum OhtaniRule {
        ONE_PLAYER, TWO_PLAYERS, MIXED
    }

    // Fields
    private Long id;
    private League league;

        // Eligibility
        private boolean positionEligibility;
        private boolean multiPositionEnabled;
        private boolean prospectEligibility;
        private boolean rookieStatusFilter;

        // Pool
        private boolean mlbOnly;
        private boolean mlbPlusProspects;
        private boolean minorLeaguePlayers;
        private boolean freeAgents;
        private OhtaniRule ohtani;
        private Division division;

        // Injury Handling
        private boolean draftInjuredPlayers;
        private boolean autoInjuryRisk;
        private boolean hideLongTermIL;

    // Constructors
    public LeagueSettings() { // Default conditions
        this.positionEligibility = true;
        this.multiPositionEnabled = false;
        this.prospectEligibility = true;
        this.rookieStatusFilter = true;
        this.mlbOnly = true;
        this.mlbPlusProspects = false;
        this.minorLeaguePlayers = false;
        this.freeAgents = false;
        this.ohtani = OhtaniRule.MIXED;
        this.division = Division.MIXED;
        this.draftInjuredPlayers = false;
        this.autoInjuryRisk = true;
        this.hideLongTermIL = true;
    }

    // Getters
    public Long getId() { return id; }
    public League getLeague() { return league; }
    public boolean isPositionEligibility() { return positionEligibility; }
    public boolean isMultiPositionEnabled() { return multiPositionEnabled; }
    public boolean isProspectEligibility() { return prospectEligibility; }
    public boolean isRookieStatusFilter() { return rookieStatusFilter; }
    public boolean isMlbOnly() { return mlbOnly; }
    public boolean isMlbPlusProspects() { return mlbPlusProspects; }
    public boolean isMinorLeaguePlayers() { return minorLeaguePlayers; }
    public boolean isFreeAgents() { return freeAgents; }
    public OhtaniRule getOhtaniRule() { return ohtani; }
    public Division getDivision() { return division; }
    public boolean isDraftInjuredPlayers() { return draftInjuredPlayers; }
    public boolean isAutoInjuryRisk() { return autoInjuryRisk; }
    public boolean isHideLongTermIL() { return hideLongTermIL; }

    // Setters
    public void setId(Long id) { this.id = id; }
    public void setLeague(League league) { this.league = league; }
    public void setPositionEligibility(boolean positionEligibility) { this.positionEligibility = positionEligibility; }
    public void setMultiPositionEnabled(boolean multiPositionEnabled) { this.multiPositionEnabled = multiPositionEnabled; }
    public void setProspectEligibility(boolean prospectEligibility) { this.prospectEligibility = prospectEligibility; }
    public void setRookieStatusFilter(boolean rookieStatusFilter) { this.rookieStatusFilter = rookieStatusFilter; }
    public void setMlbOnly(boolean mlbOnly) { this.mlbOnly = mlbOnly; }
    public void setMlbPlusProspects(boolean mlbPlusProspects) { this.mlbPlusProspects = mlbPlusProspects; }
    public void setMinorLeaguePlayers(boolean minorLeaguePlayers) { this.minorLeaguePlayers = minorLeaguePlayers; }
    public void setFreeAgents(boolean freeAgents) { this.freeAgents = freeAgents; }
    public void setOhtaniRule(OhtaniRule ohtani) { this.ohtani = ohtani; }
    public void setDivision(Division division) { this.division = division; }
    public void setDraftInjuredPlayers(boolean draftInjuredPlayers) { this.draftInjuredPlayers = draftInjuredPlayers; }
    public void setAutoInjuryRisk(boolean autoInjuryRisk) { this.autoInjuryRisk = autoInjuryRisk; }
    public void setHideLongTermIL(boolean hideLongTermIL) { this.hideLongTermIL = hideLongTermIL; }

}

File: backend/src/models/playerSettings.ts
import { DataTypes, Model, Association} from 'sequelize';
import sequelize from '../config/database';
import League from './league';

// Enums
export enum Division {
    AL = 'AL', NL = 'NL', MIXED = 'MIXED'
}

export enum OhtaniRule {
    ONE_PLAYER = 'ONE_PLAYER', TWO_PLAYERS = 'TWO_PLAYERS', MIXED = 'MIXED'
}

class PlayerSettings extends Model {
    // Fields
    public id!: number;
    public positionEligibility!: boolean;
    public multiPositionEnabled!: boolean;
    public prospectEligibility!: boolean;
    public rookieStatusFilter!: boolean;
    public mlbOnly!: boolean;
    public mlbPlusProspects!: boolean;
    public minorLeaguePlayers!: boolean;
    public freeAgents!: boolean;
    public draftInjuredPlayers!: boolean;
    public autoInjuryRisk!: boolean;
    public hideLongTermIL!: boolean;
    public ohtaniRule!: OhtaniRule;
    public division!: Division;
    public league_id!: number;
    public league!: League;

    // Associations
    public static associations: {
        league: Association<PlayerSettings, League>;
    }

    public static associate (models: any) {
        PlayerSettings.belongsTo(models.League, { foreignKey: 'league_id', as: 'league' });
    }
}

PlayerSettings.init({
    id: {
        type: DataTypes.BIGINT,
        primaryKey: true,
        autoIncrement: true
    },
    positionEligibility: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    multiPositionEnabled: {
        type: DataTypes.BOOLEAN,
        defaultValue: false
    },
    prospectEligibility: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    rookieStatusFilter: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    mlbOnly: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    mlbPlusProspects: {
        type: DataTypes.BOOLEAN,
        defaultValue: false
    },
    minorLeaguePlayers: {
        type: DataTypes.BOOLEAN,
        defaultValue: false
    },
    freeAgents: {
        type: DataTypes.BOOLEAN,
        defaultValue: false
    },
    draftInjuredPlayers: {
        type: DataTypes.BOOLEAN,
        defaultValue: false
    },
    autoInjuryRisk: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    hideLongTermIL: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    ohtaniRule: {
        type: DataTypes.ENUM(...Object.values(OhtaniRule)),
        defaultValue: OhtaniRule.MIXED
    },
    division: {
        type: DataTypes.ENUM(...Object.values(Division)),
        defaultValue: Division.MIXED
    }
}, {
    sequelize,
    tableName:'player_settings'
});

export default PlayerSettings;